iothub/device/src/Authentication/IotHubConnectionStringBuilder.cs

```diff
@@ -65,40 +56,40 @@ public IotHubConnectionStringBuilder(string iotHubConnectionString)
 
             // We'll parse the connection string and use that to build an auth method
             ExtractPropertiesFromConnectionString(iotHubConnectionString);
-            AuthenticationMethod = AuthenticationMethodFactory.GetAuthenticationMethod(this);
+            AuthenticationMethod = AuthenticationMethodFactory.GetAuthenticationMethodFromConnectionString(this);
 
             Validate();
         }
 
         /// <summary>
         /// Gets or sets the value of the fully-qualified DNS hostname of the IoT hub service.
         /// </summary>
-        public string HostName { get; internal set; }
+        public string HostName { get; private set; }
 
         /// <summary>
         /// Gets the optional name of the gateway to connect to
         /// </summary>
-        public string GatewayHostName { get; internal set; }
```

Comment:
These properties are expected to be set only through the constructor

Reply:
readonly then?

Reply:
The constructor itself calls a static method to do most of the settings. So, I made the setter private.